src/Illuminate/Routing/ResponseFactory.php

```diff
@@ -87,6 +88,33 @@ public function view($view, $data = [], $status = 200, array $headers = [])
         return $this->make($this->view->make($view, $data), $status, $headers);
     }
 
+    /**
+     * Create a new compressed JSON response instance. Requires php-zip extension.
+     *
+     * @param  array|string  $data
+     * @param  int  $status
+     * @param  array  $headers
+     * @return Response
+     */
+    public function compressedJson(array|string $data = [], $status = 200, array $headers = [])
+    {
+        if (! extension_loaded('zip')) {
+            throw new RuntimeException('Gzip extension is not available.');
+        }
+
+        if (is_array($data)) {
+            $data = json_encode($data);
+        }
+
+        $data = gzencode($data, 9);
```

Comment:
Would you consider dropping the hard coded `$level` parameter? 

I usually never use a level greater than 5 as the output difference is negligible, but the CPU work required is not. Specially for large files.

If one prefer to always use level 9, they could configure it on their server's `php.ini` (`zlib.output_compression_level`)